docs/requirements/hardware-proposal.md

```diff
@@ -0,0 +1,197 @@
+* CPU
+  * ARM Cortex-M4 MCU with FPU(80MHz) with low power
+* Memory
+  * Flash: 1MiB
+  * RAM: 128KiB
+* Interfaces I2C, SPI, UART, USB
+* Voltage range: 1.71 to 5.5V
+
+#### SPI Slave Frequency
+
+* SPI clock frequency:
+  * Slave mode receiver, 1.7 < VDD < 3.6V: 40MHz
+  * Full duplex/Slave mode transmit, 2.7 < VDD < 3.6V: 26MHz
+  * Full duplex/Slave mode transmit, 1.71 < VDD < 3.6V: 16MHz
+
+#### Cryptographic accelerators
+
+* Cryptography engine
+
+#### Software tools
+
+* RTOS Support (e.g. Zephyr)
+* STM32CubeIDE [STM32CubeIDE](https://www.st.com/en/development-tools/stm32cubeide.html)
+
+### STM32G474RE Nucleo board
+
+* [STM32G474 Nucleo Board Documentation](https://www.st.com/en/evaluation-tools/nucleo-g474re.html)
+* CPU
+  * ARM Cortex-M4 MCU (170MHz)
+* Memory:
+  * Flash: 512KiB
+  * RAM: 96KiB
+* Interfaces I2C, SPI, UART, USB
+* Voltage range: 1.71 to 5.5V
+
+#### SPI Slave Frequency
+
+* Max SPI clock frequency:
+  * Slave, receiver mode,  1.71V < Vdd < 3.6V: 50MHz
+  * Slave mode transmitter, with full duplex, 2.7V < Vdd < 3.6V: 41MHz
+  * Slave mode Transmitter, with full duplex, 1.7V < Vdd < 3.6V: 27MHz
+
+#### Cryptographic accelerators
+
+* Mathematical hardware accelerators
+  * CORDIC for trigonometric functions acceleration
+  * FMAC: filter mathematical accelerator
+
+#### Software tools
+
+* RTOS Support (e.g. Zephyr)
+* STM32CubeIDE [STM32CubeIDE](https://www.st.com/en/development-tools/stm32cubeide.html)
+
+## Requirements summary
+
+* The goal for this task is to collect and summarize, in convenient form,
+functionalities of the boards available in market, with interesting
+functionalities (communication, math/crypto accelerators). This document
+should aim to decide, which solution best fits the requirements.
+* if feature is present/useful for our solution, then factor is +1
+* if feature is not present/has limited functionality factor is 0
+  * If feature is hard to implement, score is -1
+* if the property could add extra feature (e.g. ease of implementation,
+add flexibility in further extension of the project) then multiply weight by 2. 
+
+|Property			|Reference val.	|Infineon CY8C6xxxx		|Thing Plus - EOS S3	|STM32L476RG Nucleo	|STM32G474RE Nucleo	|
+|:------------------------------|:-------------:|:------------------------------|:----------------------|:----------------------|:----------------------|
+|Supply voltage			|1.8-3.3[V]	|1.7-5.5V			|1.62-3.63		|1.71-3.6		|1.71-3.6		|
+|Current  			|<250[mA]	|				|			|			|			|
+|NV Memory 			|> 2 * 16[KiB]	|384-1856 KiB		(+1)	|16 Mb		(+2)	|1MiB		(+1)	|512KiB		(+1)	|
+|RAM Memory 			|> 16[KiB]	|176-944KiB		(+1)	|512KiB		(+2)	|128KiB		(+1)	|96KiB		(+1)	|
+|Clock Frequency 		|[MHz]		|2 cores, M4:150, M0:100  (+2)	|80		(+1)	|80		(+1)	|170		(+2)	|
+|SPI Frequency 			|[MHz]		|(no data)		(0)	|(no data)	(0)	|26		(+1)	|41		(+2)	|
+|OSSoftware support		|		|(no data)		(0)	|FPGA-based	(+1)	|yes		(+1)	|yes		(+2)	|
```

Comment:
Why each Nucleo board has different number of points? Their description above w.r.t. software is identical.